.github/workflows/pre-submit.cli.yml

```diff
@@ -18,9 +18,10 @@ jobs:
         uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
 
       - name: setup-go
-        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+        uses: actions/setup-go@v5.0.0
```

Comment:
At first I figured since it's a test dependency, and these are official github actions, it would be safe. But you're right. We could rely on the deps bot to auto-upgrade.

Reply:
I agree there's a bit of UX friction too so I'm not entirely against it :( In fact see https://github.com/ossf/scorecard/issues/2018.
But being humans I'm sure we'll forget if we have _some_ workflows pinned and some _not_ based on the triggered events